sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/text/TextTableProvider.java

```diff
@@ -141,6 +156,51 @@ public LinesReadConverter() {}
     }
   }
 
+  /** Read-side converter for {@link TextJsonTable} with format {@code 'json'}. */
+  @VisibleForTesting
+  static class JsonToRow extends PTransform<PCollection<String>, PCollection<Row>>
+      implements Serializable {
+
+    private ObjectMapper objectMapper;
+    private Schema schema;
+
+    @VisibleForTesting
+    public JsonToRow(ObjectMapper objectMapper, Schema schema) {
+      this.objectMapper = objectMapper;
+      this.schema = schema;
+    }
+
+    @Override
+    public PCollection<Row> expand(PCollection<String> input) {
+      return input
+          .apply(
+              ParDo.of(
+                  new DoFn<String, Row>() {
+                    @ProcessElement
+                    public void processElement(ProcessContext context) {
+                      context.output(jsonToRow(objectMapper, context.element()));
+                    }
+                  }))
+          .setRowSchema(schema);
+    }
+  }
+
+  /** Write-side converter for {@link TextJsonTable} with format {@code 'json'}. */
+  @VisibleForTesting
+  static class RowToJson extends PTransform<PCollection<Row>, PCollection<String>>
+      implements Serializable {
+    private ObjectMapper objectMapper;
+
+    public RowToJson(ObjectMapper objectMapper) {
+      this.objectMapper = objectMapper;
+    }
+
+    @Override
+    public PCollection<String> expand(PCollection<Row> input) {
+      return input.apply(ToJson.of());
```

Comment:
objectMapper is unused?

Reply:
objectMapper will not be used here, it simply applies _ToJson_